packages/expo/android/src/main/java/expo/modules/ReactNativeHostWrapperBase.kt

```diff
@@ -109,5 +110,15 @@ open class ReactNativeHostWrapperBase(
     return method!!.invoke(host) as T
   }
 
+  /**
+   * Inject the @{ReactInstanceManager} from the wrapper to the wrapped host.
+   * In case the wrapped host to call `getReactInstanceManager` inside its methods.
+   */
+  fun injectHostReactInstanceManager(reactInstanceManager: ReactInstanceManager) {
+    val mReactInstanceManagerField = ReactNativeHost::class.java.getDeclaredField("mReactInstanceManager")
+    mReactInstanceManagerField.isAccessible = true
+    mReactInstanceManagerField.set(host, reactInstanceManager)
+  }
```

Comment:
Are we sure that the host won't access the instance manager before we replace it? If not, that may cause a lot of problems in the future.

Reply:
> Are we sure that the host won't access the instance manager before we replace it? If not, that may cause a lot of problems in the future.

tbh i don't 100% confident that the assumption will not change in the future. but as far as i can tell, it would be safe right now in the ReactNativeHost of the default react-native template. moreover, if we don't replace it, it will introduce duplicated instance manager and that's also not we want.

i would say that the ReactNativeHostWrapper is not perfect though. if there's other broken findings, i'll try to fix them one by one.